src/register.cc

```diff
@@ -155,32 +158,60 @@ NCCL_API(ncclResult_t, ncclCommRegister, const ncclComm_t comm, void* buff, size
 ncclResult_t ncclCommRegister_impl(const ncclComm_t comm, void* buff, size_t size, void** handle) {
   NCCLCHECK(CommCheck(comm, "ncclCommRegister", "comm"));
   if (comm->checkPointers) NCCLCHECK(CudaPtrCheck(buff, comm, "buff", "ncclCommRegister"));
-  NCCLCHECK(ncclRegister(comm, buff, size, handle));
+  #ifdef ENABLE_MSCCLPP
+    if (comm->mscclCompatible && size > 0 && (size & 31) == 0 && size <= comm->mscclpp_threshold){
+      bool isManagedBuffer = false; 
+      CUDACHECK(hipPointerGetAttribute(&isManagedBuffer, HIP_POINTER_ATTRIBUTE_IS_MANAGED, const_cast<void*>(buff)));
+      if(!isManagedBuffer){
+        INFO(NCCL_INIT, "MSCCL++: ncclCommRegister");
+        NCCLCHECK(mscclpp_ncclCommRegister(comm->mscclpp_comm, buff, size, handle));
+      }
+      else{
+        WARN("MSCCL++: Cannot register user-buffers on managed memory");
+      }
+    }
+    else
+    #endif
+    {
+       INFO(NCCL_INIT, "RCCL: ncclCommRegister");
+      NCCLCHECK(ncclRegister(comm, buff, size, handle));
+    }
```

Comment:
```suggestion
  #ifdef ENABLE_MSCCLPP
    if (comm->mscclCompatible && size > 0 && (size & 31) == 0 && size <= comm->mscclpp_threshold){
      bool isManagedBuffer = false; 
      CUDACHECK(hipPointerGetAttribute(&isManagedBuffer, HIP_POINTER_ATTRIBUTE_IS_MANAGED, const_cast<void*>(buff)));
      if(!isManagedBuffer){
        INFO(NCCL_INIT, "MSCCL++: ncclCommRegister");
        NCCLCHECK(mscclpp_ncclCommRegister(comm->mscclpp_comm, buff, size, handle));
      }
      else{
        WARN("MSCCL++: Cannot register user-buffers on managed memory");
      }
      return ncclSuccess;
    }
  #endif
  INFO(NCCL_INIT, "RCCL: ncclCommRegister");
  NCCLCHECK(ncclRegister(comm, buff, size, handle));
```
Proposing this to reduce the diff and avoid future merge conflicts.